metadata-models/src/main/pegasus/com/linkedin/schema/ForeignKeyConstraint.pdl

```diff
@@ -0,0 +1,37 @@
+namespace com.linkedin.schema
+import com.linkedin.common.Urn
+
+/**
+ * Description of a foreign key constraint in a schema.
+ */
+record ForeignKeyConstraint {
+  /**
+   * Name of the constraint, likely provided from the source
+   */
+  name: string
+
+  /**
+   * Fields the constraint maps to on the foreign dataset
+   */
+  @Relationship = {
+    "/*": {
+      "name": "ForeignKeyTo",
+      "entityTypes": [ "schemaField" ]
+    }
+  }
+  foreignFields: array[Urn]
+
+  /**
+   * Fields the constraint maps to on the source dataset
+   */
+  sourceFields: array[Urn]
+
+  /**
+   * Reference to the foreign dataset for ease of lookup
+   */
+  @Relationship = {
+    "name": "ForeignKeyToDataset",
+    "entityTypes": [ "dataset" ]
+  }
+  foreignDataset: Urn
```

Comment:
Consider placing this at the top of this model

Reply:
I think it makes sense where it is- this is just for sake of easy reference.